src/Markdig.Wpf/MarkdownViewer.cs

```diff
@@ -79,9 +99,88 @@ private static void PipelineChanged(object sender, DependencyPropertyChangedEven
             control.RefreshDocument();
         }
 
+        public MarkdownViewer()
+        {
+            CommandBindings.Add(new CommandBinding(Commands.Navigate, NavigateCommandExecuted));
+        }
+
+        private void NavigateCommandExecuted(object sender, ExecutedRoutedEventArgs e)
+        {
+            string url = e.Parameter?.ToString() ?? "";
+            if (url.Length > 1 && url[0] == '#')
+            {
+                string anchorName = url.Substring(1);
+                e.Handled = NavigateTo(anchorName);
+            }
+        }
+
         protected virtual void RefreshDocument()
         {
             Document = Markdown != null ? Wpf.Markdown.ToFlowDocument(Markdown, Pipeline ?? DefaultPipeline) : null;
         }
+
+        public override void OnApplyTemplate()
+        {
+            docViewer =  GetTemplateChild("PART_DocViewer") as FlowDocumentScrollViewer;
+            
+            base.OnApplyTemplate();
+        }
+
+        public bool NavigateTo(string anchorName)
+        {
+            if (Document == null)
+                throw new InvalidOperationException("No rendered content found");
+            
+            foreach (var block in Document.Blocks)
+            {
+                string blockAnchorName = GetAnchorName(block);
+                if (String.Equals(blockAnchorName, anchorName, StringComparison.OrdinalIgnoreCase))
+                {
+                    return ScrollIntoView(block);
+                }
+            }
+
+            return false;
+        }
+
+        private bool ScrollIntoView(Block block)
```

Comment:
I pulled this concept into my own customizations. This scroll into view method occasionally is fine in theory, but the document start location literally shift around. The following is normalizing the problem I am having:
```CSharp
var objTop = block.ContentStart.GetCharacterRect(LogicalDirection.Forward).Top;
var docTop = _viewer.Document.ContentStart.GetCharacterRect(LogicalDirection.Forward).Top;
scrollViewer.ScrollToVerticalOffset(Math.Abs(docTop) + objTop);
```